### PR TITLE
SourceMap generation reads internal sources instead of from files

### LINKED ISSUE #831: Wrong paths for node modules with cli --sourceMap option
### Problem:
When importing an AS node module, the paths for the sourceMaps are not generated correctly. The cli is looking for files in the `~lib/` folder of `node_modules/assemblyscript/` instead of the `node_modules/MODULE_NAME/` folder.

The generated paths are correct when using a relative import:
import { u128 } from "../node_modules/bignum/assembly/integer/u128";  :white_check_mark:
import { u128 } from "bignum";  :x: 

### How to recreate:

Create a new starter project & add bignum (or any other AS node module) as a dependency:

```
$ mkdir new-project && cd new-project
$ yarn add AssemblyScript/assemblyscript -D
$ npx asinit .
$ yarn add MaxGraey/bignum.wasm
```
**And then import & use bignum in assembly/index.ts and compile the project**
```
import { u128 } from "bignum";

export function add(): void {
  const test = u128.from<i32>(0);
}
```
`$ yarn run asbuild:untouched`

**Which results in:**
```
> asc assembly/index.ts -b build/untouched.wasm -t build/untouched.wat --sourceMap --validate --debug

ERROR: Source file '~lib/bignum/utils.ts' not found.
    at sourceMap.sources.forEach (/Users/stefaniedoll/Parity/ParityJS/bignum-assemblyscript/node_modules/assemblyscript/cli/asc.js:723:31)
    at Array.forEach (<anonymous>)
    at Object.main (/Users/stefaniedoll/Parity/ParityJS/bignum-assemblyscript/node_modules/assemblyscript/cli/asc.js:707:29)
    at Object.<anonymous> (/Users/stefaniedoll/Parity/ParityJS/bignum-assemblyscript/node_modules/assemblyscript/bin/asc:21:26)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:754:12)
ERROR: Source file '~lib/bignum/globals.ts' not found.
    at sourceMap.sources.forEach (/Users/stefaniedoll/Parity/ParityJS/bignum-assemblyscript/node_modules/assemblyscript/cli/asc.js:723:31)
    at Array.forEach (<anonymous>)
    at Object.main (/Users/stefaniedoll/Parity/ParityJS/bignum-assemblyscript/node_modules/assemblyscript/cli/asc.js:707:29)
    at Object.<anonymous> (/Users/stefaniedoll/Parity/ParityJS/bignum-assemblyscript/node_modules/assemblyscript/bin/asc:21:26)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:754:12)
ERROR: Source file '~lib/bignum/integer/u128.ts' not found.
    at sourceMap.sources.forEach (/Users/stefaniedoll/Parity/ParityJS/bignum-assemblyscript/node_modules/assemblyscript/cli/asc.js:723:31)
    at Array.forEach (<anonymous>)
    at Object.main (/Users/stefaniedoll/Parity/ParityJS/bignum-assemblyscript/node_modules/assemblyscript/cli/asc.js:707:29)
    at Object.<anonymous> (/Users/stefaniedoll/Parity/ParityJS/bignum-assemblyscript/node_modules/assemblyscript/bin/asc:21:26)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:754:12)
```

### PR DESCRIPTION
**Pull request by willemneal**
fixes #831
- changed asinit to executable.

### CODE REVIEW

File: cli/asc.js [hunk 1]
```diff
@@ -704,27 +704,16 @@ exports.main = function main(argv, options, callback) {
...
-              }
-            } else {
-              text = readFile(name, baseDir);
-            }
-            if (text === null) {
-              return callback(Error("Source file '" + name + "' not found."));
+          console.log(sourceMap.sources)
+          for (const index in sourceMap.sources) {
+            let internalPath = sourceMap.sources[index].replace(/\.ts$/, "");
+            let source = program.sources.find(source => source.internalPath === internalPath);
```
Comment (dcodeIO) [inline cli/asc.js]: Calls into the compiler without using a C-like API, hence is non-portable once we can compile the compiler itself to Wasm.

Comment (willemneal) [reply cli/asc.js]: Ah I had a feeling this was too easy.

Comment (MaxGraey) [reply cli/asc.js]: But wait. This already happened before: https://github.com/AssemblyScript/assemblyscript/blob/f9230cd2ab8a0044a844b2d626b742a3c3e86558/cli/asc.js#L522

**Comment by dcodeIO** [comment]
Should be fixed in https://github.com/AssemblyScript/assemblyscript/pull/846

**Pull request closed without merging**